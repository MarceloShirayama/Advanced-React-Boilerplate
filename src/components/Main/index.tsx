export function Main() {
  return (
    <main>
      <h1>Welcome to React Avan√ßado!</h1>
    </main>
  )
}
