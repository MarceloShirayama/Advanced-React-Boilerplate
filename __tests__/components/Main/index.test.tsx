import { render, screen } from '@testing-library/react'

import { Main } from 'src/components/Main'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    const heading = screen.getByRole('heading', {
      level: 1,
      name: /React AvanÃ§ado!/i
    })

    expect(heading).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
