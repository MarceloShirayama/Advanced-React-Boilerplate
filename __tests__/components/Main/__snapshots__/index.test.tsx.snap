// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="sc-e07d1157-0 ioLEpy"
>
  <img
    alt="Imagem de um átomo com fundo rosa e React Avançado escrito ao lado"
    class="sc-e07d1157-1 cSfXQT"
    src="/img/logo.svg"
  />
  <h1
    class="sc-e07d1157-2 kwakBU"
  >
    React Avançado!
  </h1>
  <h2
    class="sc-e07d1157-3 eLlIIP"
  >
    Typescript, React, NextJS e Styled Components
  </h2>
  <img
    alt="Um desenvolvedor de frente para a tela com código"
    class="sc-e07d1157-4 iWLhGc"
    src="/img/hero-illustration.svg"
  />
</main>
`;
